be/src/exec/parquet_reader.cpp

```diff
@@ -0,0 +1,433 @@
+    }
+}
+
+void ParquetReaderWrap::close() {
+    _parquet->Close();
+}
+
+Status ParquetReaderWrap::size(int64_t* size) {
+    _parquet->GetSize(size);
+    return Status::OK;
+}
+
+void ParquetReaderWrap::fill_slot(Tuple* tuple, SlotDescriptor* slot_desc, MemPool* mem_pool, const uint8_t* value, int32_t len) {
+    if (len != 0) {
+        tuple->set_not_null(slot_desc->null_indicator_offset());
+        void* slot = tuple->get_slot(slot_desc->tuple_offset());
+        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
+        str_slot->ptr = reinterpret_cast<char*>(mem_pool->allocate(len));
+        memcpy(str_slot->ptr, value, len);
+        str_slot->len = len;
+    }
+    return;
+}
+
+Status ParquetReaderWrap::column_indices(const std::vector<SlotDescriptor*>& tuple_slot_descs)
+{
+    _column_ids.clear();
+    for (auto slot_desc : tuple_slot_descs) {
+        // Get the Column Reader for the boolean column
+        auto iter = _map_column.find(slot_desc->col_name());
+        if (iter == _map_column.end()) {
+            std::stringstream str_error;
+            str_error << "Invalid Column Name:" << slot_desc->col_name();
+            return Status(str_error.str());
+        }
+        _column_ids.emplace_back(iter->second);
+    }
+    return Status::OK;
+}
+
+void ParquetReaderWrap::set_field_null(Tuple* tuple, const SlotDescriptor* slot_desc) {
+    if (!slot_desc->is_nullable()) {
+        throw parquet::ParquetException("Null is not allowed, but Parquet field is NULL.");
+    }
+    tuple->set_null(slot_desc->null_indicator_offset());
+}
+
+Status ParquetReaderWrap::read_record_batch(const std::vector<SlotDescriptor*>& tuple_slot_descs, bool* eof) {
+    if (_current_line_of_group == 0) {// the first read
+        RETURN_IF_ERROR(column_indices(tuple_slot_descs));
+        // read batch
+        _reader->GetRecordBatchReader({_current_group}, _column_ids, &_rb_batch);
+        arrow::Status status = _rb_batch->ReadNext(&_batch);
+        if (!status.ok()) {
+            LOG(WARNING) << "The first read record. " << status.ToString();
+            throw parquet::ParquetException(status.ToString());
+        }
+    } else if (_current_line_of_group >= _rows_of_group) {// read next row group
+        _current_group++;
+        if (_current_group >= _total_groups) {// read completed.
+            _column_ids.clear();
+            *eof = true;
+            return Status::OK;
+        }
+        _current_line_of_group = 0;
+        _rows_of_group = _file_metadata->RowGroup(_current_group)->num_rows(); //get rows of the current row group
+        // read batch
+        _reader->GetRecordBatchReader({_current_group}, _column_ids, &_rb_batch);
+        arrow::Status status = _rb_batch->ReadNext(&_batch);
+        if (!status.ok()) {
+            LOG(WARNING) << status.ToString();
+            throw parquet::ParquetException(status.ToString());
```

Comment:
change throw exception to return status. we don't use exception in our CPP files